### ISSUE TITLE
Support for Code 128 Barcodes

### ISSUE THREAD

**Issue by bredowmax** [openfoodfacts/openfoodfacts-ios#563]
**Summary:** 
Some products (mostly freshly packaged products with varying weight or price) have Code 128 barcodes that the OFF app does not currently support. The Scandit app does find it

**Screenshots:** 
<img width="1050" alt="Screenshot 2020-02-01 at 16 08 58" src="https://user-images.githubusercontent.com/30573532/73599344-6553b780-4510-11ea-8278-eb07fa76d4e2.png">

**What is Code 128?**
https://en.wikipedia.org/wiki/Code_128

**Slack**
https://openfoodfacts.slack.com/archives/C02SSQ02N/p1580590985010200

**Comment by bredowmax** [comment]
Another example, this time in Germany: https://world.openfoodfacts.org/product/2312251310191757006523

**Comment by aleene** [comment]
This is a "feature" of the barcodescanner that OFF uses. Or does is this not seen as a valid code? My app does recognizes this code.

**Comment by bredowmax** [comment]
![F474BEC8-2BFD-43F6-B3A1-35FE81E59BAF](https://user-images.githubusercontent.com/30573532/73607673-17c66180-4587-11ea-872f-f0be915c78a9.png)

If I wanted to add the product manually to OFF - what is the correct code? The 16 digits?

**Comment by bredowmax** [comment]
True, the Edeka example works on my iOS OFF. The WholeFoods one, however, does not get recognised by OFF (Scandit says both codes are a Code 128 code)

**Comment by bredowmax** [comment]
The Edeka example has 22 digits, but the Whole Foods example has 27 digits

**Comment by aleene** [comment]
Maybe there is a number of digits check.

**Comment by teolemon** [comment]
https://developer.apple.com/documentation/vision/vnbarcodesymbology ?

**Comment by bredowmax** [comment]
Thanks @teolemon for the link. Yes, Code128 is on this list! As @aleene pointed out, Code128 is indeed supported - but apparently they can vary in length. A code with 27 digits was unsupported, so it could be a failed checking number test?

**Comment by aleene** [comment]
Just checked: Code128 is supported in the IOS app and the Pastrami code is correctly read.

It is also shown in the history.

So for me there is no issue.
